### PR TITLE
Example period histogram plots

### PR DESCRIPTION
**Pull request by mcoughlin**
Period histograms based on the feature set.

### CODE REVIEW

File: scope.py [hunk 1]
```diff
@@ -278,6 +324,49 @@ def doc(self):
...
+                features = self._get_features(positions=positions)
+
+                if len(features) == 0:
+                    print(f'No features for {golden_set_name}')
+                    continue
+
+                if golden_set_name in period_limits:
+                    limits = period_limits[golden_set_name]
+                else:
+                    limits = None
```
Comment (dmitryduev) [inline scope.py:356]:
```suggestion
                limits = period_limits.get(golden_set_name)
```

File: scope.py [hunk 2]
```diff
@@ -278,6 +324,49 @@ def doc(self):
...
+        period_limits = {'cepheid': [1.0, 100.0],
+                         'delta_scuti': [0.03, 0.3],
+                         'beta_lyr': [0.3, 25],
+                         'rr_lyr': [0.2, 1.0],
+                         'w_uma': [0.2, 0.8]}
+        period_loglimits = {'cepheid': True,
+                            'delta_scuti': False,
+                            'beta_lyr': True,
+                            'rr_lyr': False,
+                            'w_uma': False}
```
Comment (dmitryduev) [inline scope.py:336]: Lint, please

File: scope/utils.py [hunk 3]
```diff
@@ -102,6 +102,60 @@ def plot_light_curve_data(
...
+    plt.rc("text", usetex=True)
+
+    # make figure
+    fig, ax = plt.subplots(figsize=(6, 6))
+    if title is not None:
+        fig.suptitle(title, fontsize=24)
+
+    if limits is not None:
+        if loglimits:
+            edges = np.logspace(np.log10(limits[0]), np.log10(limits[1]),
```
Comment (dmitryduev) [inline scope/utils.py:126]: Lint here and below. How come the linter didn't complain?

**Review by dmitryduev — APPROVED** [review]
Nice @mcoughlin. Could you please add a screenshot or two to the description? Otherwise LGTM

File: scope.py (hunk 2 above)
Comment (mcoughlin) [reply scope.py:336]: What exactly is failing? I ran the pre-commit install and so on.

File: scope/utils.py (hunk 3 above)
Comment (mcoughlin) [reply scope/utils.py:126]: again, what is failing?

**Comment by mcoughlin** [comment]
period__cepheid

Plots are simple. Can be fancied up as people like down the line.

File: scope.py [hunk 4]
```diff
@@ -350,10 +350,7 @@ def doc(self):
                     print(f'No features for {golden_set_name}')
                     continue
 
-                if golden_set_name in period_limits:
-                    limits = period_limits[golden_set_name]
-                else:
-                    limits = None
+                limits = period_limits.get(golden_set_name)
 
                 if golden_set_name in period_loglimits:
```
Comment (dmitryduev) [inline scope.py:355]: Also, same as above, please.

File: scope/utils.py [hunk 5]
```diff
@@ -102,6 +102,60 @@ def plot_light_curve_data(
...
+    save: Optional[str] = None,
+):
+    """Plot a histogram of periods for the sample
+
+    """
+    # plot the H-R diagram for 1 M stars within 200 pc from the Sun
+    plt.rc("text", usetex=True)
+
+    # make figure
+    fig, ax = plt.subplots(figsize=(6, 6))
```
Comment (dmitryduev) [inline scope/utils.py:120]: Make it (16, 5) to be on par with the other plots?

Comment (mcoughlin) [reply scope/utils.py:120]: That's only true of lightcurves. gaia is (6,6) for example. Want me to still do 16, 5?

Comment (dmitryduev) [reply scope/utils.py:120]: Whatever is more pleasing to your experienced eye (mine are not).

File: scope.py (hunk 2 above)
Comment (dmitryduev) [reply scope.py:336]: I accidentally introduced a bug in the pre-commit hook config in a previous pr. Just fixed, please merge in main, linter will work then.

File: scope/utils.py (hunk 5 above)
Comment (mcoughlin) [reply scope/utils.py:118]: I think smaller is OK for the histograms here (we don't have much data).

**Review by dmitryduev — APPROVED** [review]
LGTM!

**Review by przemekmroz — APPROVED** [review]

**Pull request merged**